### PR TITLE
CI: implement tests for count matrices

### PR DESCRIPTION
**Pull request by BorisMuzellec**
This PR implements tests on the count matrix. More precisely:

- The `DeseqDataset` class now runs `test_valid_counts` at initialisation and throws a ValueError in case of unexpected inputs,
- Pytests now test that an error is thrown in case a `DeseqDataset` is initialised with NaN, non-integer or negative values in the count matrix,
- Pytests test the behaviour of  `DeseqDataset` and `DeseqStats` when a gene only has zero counts for all samples.

### CODE REVIEW

**Review by arthurPignetOwkin — APPROVED** [review]
Great job!

File: pydeseq2/utils.py [hunk 1]
```diff
@@ -115,6 +115,28 @@ def load_data(
...
+
+    More precisely, test that inputs are non-negative integers.
+
+    Parameters
+    ----------
+    counts_df : pandas.DataFrame
+        Raw counts. One column per gene, rows are indexed by sample barcodes.
+    """
+    if counts_df.isna().any().any():
+        raise ValueError("NaNs are not allowed in the count matrix.")
```
[the remark this answers is not in the document]
Comment (BorisMuzellec) [reply pydeseq2/utils.py:129]: Good point. I've reproduced what DESeq2 does but it would make sense to throw a warning and output NaN results for genes that have at least one NaN in the count matrix.

I'll merge this PR for now and open an issue to discuss this.
